hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/S3SecretManagerImpl.java

```diff
@@ -64,6 +69,10 @@ public String getSecretString(String awsAccessKey)
         "awsAccessKeyId cannot be null or empty.");
     LOG.trace("Get secret for awsAccessKey:{}", awsAccessKey);
 
+    S3SecretValue cacheValue = s3SecretCache.get(awsAccessKey);
```

Comment:
Today, s3SecretCache is updated from `OMSetSecretRequest` and `S3GetSecretRequest` only. 
It's not updated on normal S3 operations (key/bucket read/write). 
Thus, there's a good chance that this cache doesn't help improve the performance or reduce the load to the external vault. 

The Secret cache should be implemented as a read-through cache (or LoadingCache in Guava term).

And IMHO, for simplicity, it can be just a `LoadingCache` directly put in `S3SecretManagerImpl`. I don't see it can be anything else (that needs the flexibility of a separate component/interface) in the future.